packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-builder-ui-workspace/dnd-wrapper.tsx

```diff
@@ -71,15 +72,21 @@ const PipelineBuilderDndWrapper = ({
     })
   );
 
+  const connectionInfo = useConnectionInfoAccess().getCurrentConnectionInfo();
+
   const handleUseCaseDropped = useCallback(
     (event: DragEndEvent) => {
       const { over } = event;
       const overId = indexFromDroppableId(String(over?.id));
       if (draggedUseCase && overId !== null) {
-        track('Aggregation Use Case Added', {
-          drag_and_drop: true,
-          stage_name: draggedUseCase.stageOperator,
-        });
+        track(
+          'Aggregation Use Case Added',
+          {
+            drag_and_drop: true,
+            stage_name: draggedUseCase.stageOperator,
+          },
+          connectionInfo
```

Comment:
If you get the info in the render before passing it, you're relying to a render method to update the value by just the virtue of something else updating the component without actually subscribing to the connection info. Even though it doesn't make much difference right now, you rather want to call the get method in place where you need the value, that guarantees that you are getting the most up to date version of connection info at the moment of the event. It also makes sure that this callback hook changes less: the return value of get method can change, but the `connectionInfoAccess` object will not

```suggestion
          connectionInfoAccess.getCurrentConnectionInfo()
```